### PR TITLE
pkp/pkp-lib#10653 remove deprecated submission functions in OAI class…

### PR DESCRIPTION
**Pull request by kaitlinnewson**
…es and templates

### CODE REVIEW

**Review by bozana — CHANGES_REQUESTED** [review]
Hi @kaitlinnewson, 
the most of the data now belongs to the publication object, so instead of using $article->getData() we would need $currentPublication->getData(). Some places have it correct, but lots of them still don't. 
Thanks a lot!

**Comment by kaitlinnewson** [comment]
Hi @bozana, ready for a second look!

File: plugins/oaiMetadataFormats/marc/OAIMetadataFormat_MARC.php [hunk 1]
```diff
@@ -33,27 +37,37 @@ class OAIMetadataFormat_MARC extends OAIMetadataFormat
...
     public function toXml($record, $format = null)
     {
+        /** @var Submission $article */
         $article = $record->getData('article');
+
+        /** @var Publication $publication */
+        $publication = $article->getCurrentPublication();
+
+        $representationDao = Application::getRepresentationDAO();
+        $representations = $representationDao->getByPublicationId($publication->getId());
```
Comment (bozana) [inline plugins/oaiMetadataFormats/marc/OAIMetadataFormat_MARC.php:47]: Why do we need this?

File: plugins/oaiMetadataFormats/marc/templates/record.tpl [hunk 2]
```diff
@@ -63,7 +63,7 @@
 		<subfield label="a">{$identifyType|escape}</subfield>
 	</varfield>{/if}
 
-	{foreach from=$article->getGalleys() item=galley}
+	{foreach from=$representations item=galley}
```
Comment (bozana) [inline plugins/oaiMetadataFormats/marc/templates/record.tpl:66]: Could we use `$publication->getData('galleys');` here?

File: plugins/oaiMetadataFormats/marcxml/templates/record.tpl [hunk 3]
```diff
@@ -25,9 +25,9 @@
 			<subfield code="a">{$journal->getData('printIssn')|escape}</subfield>
 		</datafield>
 	{/if}
-	{if $article->getStoredPubId('doi')}
+	{if $publication->getStoredPubId('doi')}
```
Comment (bozana) [inline plugins/oaiMetadataFormats/marcxml/templates/record.tpl:28]: We now use `$publication->getDoi()`.
`getStoredPubId('doi')` is old -- from the time when DOI was a plugin.

**Review by bozana — COMMENTED** [review]
Hi @kaitlinnewson, great!
Maybe one comment only -- if we can use `$publication->getData('galleys');` instead of getting the galleys anew.
And I see here is a mix of all kinds of locales in these plugins, so I created a new issue to address that https://github.com/pkp/pkp-lib/issues/10683.
Thanks a lot!

**Comment by kaitlinnewson** [comment]
@bozana fixed now!

**Pull request merged**